pkg/spi/application/types.go

```diff
@@ -0,0 +1,33 @@
+package application
+
+import (
+	"github.com/docker/infrakit/pkg/types"
+)
+
+// Type is the type of an application.  This gives hint about what struct types to map to, etc.
+// It also marks one instance of an application as of different nature from another.
+type Type string
+
+//Operation : update operation code
+type Operation int
+
+const (
+	// TypeError is the type to use for sending errors in the transport of the events.
+	TypeError = Type("error")
+
+	// ADD new resources
+	ADD Operation = iota
+	// DELETE resources
+	DELETE
+	// UPDATE resources
+	UPDATE
+	// GET resources
+	GET
+)
```

Comment:
A non-trivial example of an infrakit 'application' that I have in mind is a small daemon that listens on a tcp port, offers a RESTful api that makes it easy for infrakit to implement this:
https://github.com/kubernetes/contrib/blob/master/cluster-autoscaler/cloudprovider/cloud_provider.go#L40

So if you look at it this way, we need to have a way to give an imperative API to infrakit's inherently declarative based api (via only a verb `commit`).    We don't really need to add more plugin spi / JSON RPC implementations...  instead the 'application' can surface our internal plugin SPI in a meaningful way to other systems.

So let's chat at DockerCon more in detail about this idea.  We need to take time to get this right.